From 26a109dcb1ecde3c583c59029733cf433216b42c Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Tue, 26 Jul 2022 18:14:05 +0800
Subject: [PATCH 3527/4585] [Huawei] mm: Export static key mem_reliable

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187270

--------------------------------

Static key mem_reliable is used to check wheater memory reliable's status
in kernel's inline functions. These inline function rely on this but dirver
can not use because this symbol is not exported.
To slove this problem, export this symbol to make prepration for driver to
use memory reliable's inline function.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 include/linux/mem_reliable.h | 2 +-
 mm/mem_reliable.c            | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/mem_reliable.h b/include/linux/mem_reliable.h
index 38891cb2fa83..c3a9a4f21470 100644
--- a/include/linux/mem_reliable.h
+++ b/include/linux/mem_reliable.h
@@ -10,7 +10,7 @@
 
 #ifdef CONFIG_MEMORY_RELIABLE
 
-extern struct static_key_false mem_reliable;
+DECLARE_STATIC_KEY_FALSE(mem_reliable);
 
 extern bool reliable_enabled;
 extern bool shmem_reliable;
diff --git a/mm/mem_reliable.c b/mm/mem_reliable.c
index 15a84c0f714e..083997234e01 100644
--- a/mm/mem_reliable.c
+++ b/mm/mem_reliable.c
@@ -9,6 +9,7 @@
 #include <linux/mmzone.h>
 
 DEFINE_STATIC_KEY_FALSE(mem_reliable);
+EXPORT_SYMBOL_GPL(mem_reliable);
 
 bool reliable_enabled;
 
-- 
2.17.1

